include/support.h

```diff
@@ -35,6 +35,9 @@
 #define strncasecmp(a, b, n) _strnicmp(a, b, n)
 #endif
 
+// Convert a string to double, returning true or false depending on susccess
+bool str_to_double(const std::string& input, double &value);
```

Comment:
For most types, I would now complain about lack of `std::optional` class until we'll be able to move to C++17 (interface could be `std::optional<double> str_to_double(const std::string&)`). But in this case it's `double`, and floating-point types are special and support special values like: signaling NAN, non-signaling NAN, +infinity, -infinity, -0, +0.

I presume code calling this function does not want to handle those edge cases (with exception of 0s, of course), therefore my suggestion is: let's change this function to:

`double to_finite_double(const std::string &s)` (or simple template `double to_finite<double>(const std::string &s)` parsing errors can be communicated via [`quiet_NAN`](https://en.cppreference.com/w/cpp/types/numeric_limits/quiet_NaN), and output value will need to be verified using [`std::isfinite`](https://en.cppreference.com/w/cpp/numeric/math/isfinite) (which we probably need to use anyway to protect ourselves from `std::logic_error` which will be thrown if user will type NAN in input).

So the caller code would be turned from this:
```
	double user_value;
	if (str_to_double(str_value, user_value)) {
		…
```
into this:
```
	const double user_value = to_finite_double(str_value);
	if (std::isfinite(user_value)) {
		…
```

Reply:
Wow.. a lot has happened in the C++ world of floating point handling; thanks for all this help.  Added these changes and it's working well (I'm throwing everything at it.. characters, NANs, etc.)